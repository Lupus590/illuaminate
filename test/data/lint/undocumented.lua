--- config: (lint (only doc:undocumented))
-- Foo
-- @module

--- @type Foo
local Foo = {}

return {
  has_docs = function() end, --- This has documentation

  no_docs = function() end, -- This doesn't

  --- Has description
  no_args = function(a, b) end,

  --- Has description
  -- @param a Has description
  -- @param b
  no_arg_desc = function(a, b) end,

}
