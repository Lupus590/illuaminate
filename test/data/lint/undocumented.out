undocumented.lua:[6:13-6:14]: Type 'Foo' is exported, but has no documentation. [doc:undocumented]
   │
 6 │ local Foo = {}
   │             ^^
undocumented.lua:[11:13-11:26]: `no_docs` is exported, but has no documentation. [doc:undocumented]
    │
 11 │   no_docs = function() end, -- This doesn't
    │             ^^^^^^^^^^^^^^
undocumented.lua:[14:13-14:30]: Argument `b` is missing a description [doc:undocumented-arg]
    │
 14 │   no_args = function(a, b) end,
    │             ^^^^^^^^^^^^^^^^^^
undocumented.lua:[14:13-14:30]: Argument `a` is missing a description [doc:undocumented-arg]
    │
 14 │   no_args = function(a, b) end,
    │             ^^^^^^^^^^^^^^^^^^
undocumented.lua:[19:17-19:34]: Argument `b` is missing a description [doc:undocumented-arg]
    │
 19 │   no_arg_desc = function(a, b) end,
    │                 ^^^^^^^^^^^^^^^^^^
No errors and 5 warnings
